app/views/staff/_form_fields.html.erb

```diff
@@ -65,6 +65,21 @@
           </div>
         </div>
       </li>
+      <li class="list-group-item">
+        <div class="row">
+          <div class="col-sm-3">
+            <%= label_field form, :terms_of_service, t("activerecord.attributes.staff.terms_of_service") %>
```

Comment:
At the top of the file, you can see that it is added to the user form:
```
<%= form.simple_fields_for :user, user do |user_form| %>
```

So I assume you have to permit the `:terms_of_service` for the `user_attributes` in `staff_params`, similar to the other user attributes such as `firstname` and `lastname`.

Reply:
Did you mean like this in [staff_controller](https://github.com/openHPI/hpi-connect-portal/blob/f83b6882b711c51dbadeca363f2028fd7a51ef9b/app/controllers/staff_controller.rb)?

```ruby
def staff_params
      params.require(:staff).permit(user_attributes: [:firstname, :lastname, :email, :password, :password_confirmation, :terms_of_service])
end
```

unfortunately this did not work. Checking the checkbox still results in the same error as above

Reply:
Oh, now I see why it is still `form`. You want to validate it for `staff` and not the generic `user`.
The validation for `validates_acceptance_of` is still on the `staff` so it will also fail following my comment above.

Reply:
Please ignore my previous comments.

Are you sure this is still an issue? It worked for me locally.

Reply:
Unfortunately, it does not on my machine. Still, when I check the checkbox and submit I receive the error “Staff members terms of service muss akzeptiert werden"

Reply:
As a first possible step to debug, is the `:terms_of_service` attribute sent as part of the POST request when submitting the form? 
https://stackoverflow.com/questions/15603561/how-can-i-debug-a-http-post-in-chrome